### PR TITLE
Add API error design guide

### PR DESCRIPTION
**Pull request by drapergeek**
Hello Coggers!

One of the things we've ran into a few times is trying to properly parse and display user error messages in flywheel. This is my first pass at trying to define a basic starting style guide for how to format error messages so we can make sure display the properly.

If you have any thoughts or ideas I'd love to hear them!

### CODE REVIEW

**Comment by mpeck** [comment]
@drapergeek can you include an example of what a response would look like?

**Comment by drapergeek** [comment]
@mpeck added a few examples, let me know if that helps.

File: docs/api/errors.md [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+* `param` - if the error is directly tied to an attribute on a resource, include the
+  resource attribute. For example, if user creation fails because the user's email
+  is invalid, then `param` would be "email".
+
+  If the param is given, you can assume that the message will be displayed
+  alongside the param so it does not have to be included in the message. Eg:
+  "Email is invalid". In this case, `param` would be "email" and `message`
+  would be "is invalid".
+
+* `id` - a unique identifier to the error
```
Comment (mpeck) [inline docs/api/errors.md]: So what is the purpose of id? Does this mean that we should persist errors?

Comment (drapergeek) [inline docs/api/errors.md]: I think this is more to the type of error. For example, an error that was returned for being unable to delete the `operable` bundle.

**Comment by mpeck** [comment]
I like the idea of a style guide for things like this. I'm still a bit foggy on what the error id would be used for, but for the most part I dig it. I would like to get some other folks opinions too though.

**Comment by drapergeek** [comment]
>I like the idea of a style guide for things like this. I'm still a bit foggy on what the error id would be used for, but for the most part I dig it. I would like to get some other folks opinions too though.

I'm fine with removing the `id`. This was gathered from a few various sources.

**Pull request closed without merging**